web3/utils/abi.py

```diff
@@ -1,23 +1,42 @@
+from collections import (
+    namedtuple,
+)
+
 import re
 import itertools
 
+from cytoolz import (
+    curry,
+    pipe,
+)
+
 from eth_utils import (
+    add_0x_prefix,
     coerce_args_to_bytes,
     coerce_args_to_text,
     coerce_return_to_text,
-    to_tuple,
-    add_0x_prefix,
+    is_address,
+    is_boolean,
+    is_integer,
     is_list_like,
     is_string,
-    is_integer,
-    is_boolean,
-    is_address,
+    to_checksum_address,
+    to_tuple,
 )
 
 from eth_abi.abi import (
     process_type,
 )
 
+from web3.utils.formatters import (
+    recursive_map
+)
+
+BASE_RETURN_NORMALIZERS = [
+    lambda typ, data: (typ, to_checksum_address(data)) if typ == 'address' else (typ, data),
+    # lambda typ, data: data.decode(errors='ignore') if typ == 'string' else data,
```

Comment:
leftover comment?

Reply:
Oops, yeah, that led me down the #318 path. I'll remove it.